src/vsphere_cpi/lib/cloud/vsphere/nsxt_provider.rb

```diff
@@ -287,6 +288,108 @@ def remove_vm_from_server_pools(vm_ip)
       end
     end
 
+    def create_t1_router(edge_cluster_id, name = nil)
+      name = SecureRandom.base64(8) if name.nil?
+      router_api.create_logical_router({ :edge_cluster_id => edge_cluster_id,
+                                         :router_type => 'TIER1',
+                                         :display_name => name})
+    end
+
+    def attach_t1_to_t0(t0_router_id, t1_router_id)
+      begin
+        t0_router_port =  NSXT::LogicalRouterLinkPortOnTIER0.new({:logical_router_id => t0_router_id,
+                                                                  :resource_type => 'LogicalRouterLinkPortOnTIER0'})
+        t0_router_port = router_api.create_logical_router_port(t0_router_port)
+      rescue Exception => e
+        @logger.error("Error creating port on T0 router #{t0_router_id}. Exception: #{e}")
+        raise "Error creating port on #{t0_router_id} T0 router. Exception: #{e}"
+      end
+
+      begin
+        t0_reference = NSXT::ResourceReference.new({:target_id => t0_router_port.id,
+                                                    :target_type => 'LogicalRouterLinkPortOnTIER0',
+                                                    :is_valid => true })
+        t1_router_port = NSXT::LogicalRouterLinkPortOnTIER1.new({:linked_logical_router_port_id => t0_reference,
+                                                                 :logical_router_id => t1_router_id,
+                                                                 :resource_type => 'LogicalRouterLinkPortOnTIER1'})
+        router_api.create_logical_router_port(t1_router_port)
+      rescue Exception => e
+        @logger.error("Error creating port on T1 router #{t1_router_id} and attaching it to T0 port #{t0_router_port.id}. Exception: #{e}")
+        raise "Error creating port on T1 (#{t1_router_id}) and attaching it to T0 port #{t0_router_port.id}. Exception: #{e}"
```

Comment:
use `e.inspect` instead of just `e` so that its clear which class name error is from since a lot of exception in ruby dont have *unique* descriptions.